src/sparseml/export/registry.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+from pathlib import Path
+from typing import Any, Callable, Optional, Union
+
+from pydantic import BaseModel, Field, validator
+
+from sparseml.pytorch.image_classification.utils.helpers import (
+    create_model as create_model_ic,
+)
+from sparseml.pytorch.image_classification.utils.helpers import (
+    is_image_classification_model,
+)
+from sparsezoo.utils.registry import RegistryMixin
+
+
+__all__ = ["IntegrationHelperFunctions", "infer_integration"]
+
+
+class Integrations(Enum):
+    """
+    Holds the names of the available integrations.
+    """
+
+    image_classification = "image-classification"
+
+
+# TODO: Fold it into the functionalities of the RegistryMixin
+# when the `resolve` method is generically implemented
+def infer_integration(source_path: Union[Path, str]) -> str:
+    """
+    Infer the integration to use for exporting the model from the source_path.
+
+    :param source_path: The path to the PyTorch model to export.
+    :return: The name of the integration to use for exporting the model.
+    """
+    if is_image_classification_model(source_path):
+        return Integrations.image_classification.value
+    else:
+        raise ValueError(
+            f"Could not infer integration from source_path: {source_path}."
+            f"Please specify an argument `integration` from one of"
+            f"the available integrations: {list(Integrations)}."
+        )
+
+
+class IntegrationHelperFunctions(RegistryMixin, BaseModel):
+    """
+    Registry that maps names to helper functions
+    for creation/export/manipulation of models for a specific
+    integration.
+    """
+
+    create_model: Optional[Callable] = Field(
```

Comment:
For every function - we MUST have an expected signature (even if we're using kwargs to make a bit generic)

i.e. we define the standard that `create_model` should look something like `(model_path, **kwargs) -> Model`, `create_dummy_input(**kwargs) -> Union[List[Tensor], Tensor]`

Reply:
we could also do `(model_path, **kwargs) -> Union[Model, Optioanl[kwargs]` so we could return an arch key, the thing thing we can't do is say arch key is always output 2